meta-mender-commercial/classes/mender-closed-source-tar-utils.bbclass

```diff
@@ -0,0 +1,40 @@
+
+def mender_closed_source_srcrev_from_src_uri(d, src_uri, repo_name):
+    pref_version = d.getVar("PREFERRED_VERSION")
```

Comment:
When `PREFERRED_VERSION_pn-mender-client` is used, this value gets put into `PREFERRED_VERSION` when the recipe currently being parsed is the mender-client (`"${PN}" == "mender-client"`). However, the preferred style in Yocto is actually `PREFERRED_VERSION_mender-client`, which is not a special variable, and which is available only under that name. See example [here](https://www.yoctoproject.org/docs/latest/bitbake-user-manual/bitbake-user-manual.html#var-bb-PREFERRED_VERSION). The fact that the `pn-` version works is a combination of history and coincidence, I believe.

Therefore, if `PREFERRED_VERSION` is empty, you should also check `PREFERRED_VERSION_${PN}`, since this is the officially supported variant.

I know we are probably not consistent about this everywhere (and definitely not in mender-qa, which I just reviewed), but better to be prepared.

**Update:** I just realized that this is pre-existing, but I encourage fixing it anyway.

Reply:
Got it. I remember this :smiley: 

Thanks. I fixed it now.